doc/aspice_swe3_4/example/iceoryx_component/include/example_component/example_sample_class.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef MY_SAMPLE_CLASS_HPP_INCLUDED
+#define MY_SAMPLE_CLASS_HPP_INCLUDED
+
+#include "source/example_base_class.hpp"
+
+
+namespace example
+{
+/// @brief Short description
+/// @code
+///   minimalistic working example which uses all public methods
+/// @endcode
+/// @swcomponent example_component
+void someFreeFunction();
+
+/// @brief Forward declaration of other class
+class SomeOtherClass;
+
+/// @brief Short description
+/// @details Detailed description
+/// @code
+///
+/// @endcode
+/// @note Important note for user/developer
+/// @swcomponent cpp
+template <typename T>
+class MySampleClass : public ExampleBaseClass<T>
+{
+  public:
+    /// @brief Short description
+    /// @details Detailed description
+    /// @param[in] a Description of input parameter a
+    /// @param[in] b Description of input parameter b
+    MySampleClass(const int a, const int b);
+
+    /// @brief Short description
+    void simpleMethod() const;
+
+    /// @brief Short description
+    /// @param[in] c 	Description of input parameter c
+    /// @param[out] d 	Description of output parameter d
+    /// @return      	Description of return value
+    int complexMethod(uint32_t c, int* d);
```

Comment:
`const uint32_t c` else we get QAC++ warnings

Reply:
Additionally we have to use const for the pointer. `int complexMethod(const uint32_t c, int * const d) noexcept;`. And do not forget the `noexcept` keyword in every function and method.